### ISSUE TITLE
Don't upgrade classnames in array key strings

### ISSUE THREAD

**Issue by chillu** [silverstripe/silverstripe-upgrader#135]
## Acceptance Criteria

 * As a developer upgrading a 3.x site, I have good visibility of when the upgrader performs ambiguous actions
 * Upgrader shows explicit diff on string replacements to clearly point out the ambiguity to the developer

## Details

Before:

```php
$has_one = array('Image' => 'Image')
```

After:

```php
$has_one = array(Image::class => Image::class)
```

Expected:

```php
$has_one = array('Image' => Image::class)
```

Related: https://github.com/silverstripe/silverstripe-upgrader/issues/16

/cc @pitchandtone

# Pull Requests
- [x] https://github.com/silverstripe/silverstripe-upgrader/pull/149
- [x] https://github.com/silverstripe/silverstripe-assets/pull/204
- [ ] https://github.com/silverstripe/silverstripe-framework/pull/8741

**Comment by pitchandtone** [comment]
There's examples in code where you do things like new UploadField('Image') which then becomes new UploadField(Image::class) which is also not helpful. It might not be the only class to have this issue (maybe Member too) but has been a pain to tidy up after the script.

**Comment by Leapfrognz** [comment]
Another example: 
`UploadField:create('Image', 'Image') becomes  UploadField:create('SilverStripe\Assets\Image', 'SilverStripe\Assets\Image')`
`UploadField:create('Link', 'Link') becomes  UploadField:create('Sheadawson\Linkable\Model\Link', 'headawson\Linkable\Model\Link')`

**Comment by chillu** [comment]
Yeah, it gets pretty tricky. We could whitelist specific methods and properties which are known to accept object types (in specific positions), but that's a big list. PHP code introspection is hard.

**Comment by unclecheese** [comment]
Also `EmailField::create('Email')`.. 

We should identify SS3 classes with common names, like Image and Email, and maybe only change those strings on user confrirmation.

**Comment by chillu** [comment]
There's too many edge cases here, the best we can do is to highlight this more explicitly to devs. I've reflected this in the ACs

**Comment by maxime-rainville** [comment]
silverstripe/silverstripe-framework#8741 has not been merged yet.
